pkg/trainer/replicas.go

```diff
@@ -239,6 +239,60 @@ func (s *TFReplicaSet) CreatePodWithIndex(index int32) (*v1.Pod, error) {
 	return s.ClientSet.CoreV1().Pods(s.Job.job.ObjectMeta.Namespace).Create(pod)
 }
 
+// Delete the replicas by clean policy when the tfjob is complete or failed: CleanAll, CleanNone, CleanRunning, the default is CleanAll
+func (s *TFReplicaSet) DeleteResourcesByCleanPolicy(cleanPolicy tfv1alpha1.CleanPodPolicyType) (err error) {
+	log.Infof("DeleteResourcesByCleanPolicy for %s with CleanPodPolicyType %v", s.Job.job.ObjectMeta.Name, cleanPolicy)
+	switch cleanPolicy {
+	case tfv1alpha1.CleanUndefined, tfv1alpha1.CleanAll:
+		s.contextLogger.Infof("Apply Clean All Policy for %s", s.Job.job.ObjectMeta.Name)
+		err = s.Delete()
+	case tfv1alpha1.CleanNone:
+		s.contextLogger.Infof("Apply Clean None Policy for %s", s.Job.job.ObjectMeta.Name)
+	case tfv1alpha1.CleanRunning:
+		s.contextLogger.Infof("Apply Clean Running Pod Policy for %s", s.Job.job.ObjectMeta.Name)
+		err = s.DeleteRunningPods()
+	default:
+		s.contextLogger.Errorf("Unknown cleanPolicy %v", cleanPolicy)
+		err = fmt.Errorf("Unknown cleanPolicy %v", cleanPolicy)
```

Comment:
Shouldn't the default behavior be consistent with the current behavior which is delete all resources?
I suspect that's why the test is failing.